src/services/generateCrypto.ts

```diff
@@ -0,0 +1,7 @@
+import crypto from "crypto";
+
+export const generateCrypto: authCode = async () => {
+  return crypto.randomBytes(16).toString("hex");
+};
+
+type authCode = () => Promise<string>;
```

Comment:
To me an `authCode` is just a `string` - maybe the generating function would return a `Promise<authCode>` but the type `authCode` itself shouldn't be a Promise

Reply:
btw. this doesn't really do anything because you'd need a "branded type" to make TS differentiate between a string and an authcode here